Orange/widgets/data/owcreateclass.py

```diff
@@ -562,23 +564,30 @@ def send_report(self):
         # within the loop
         # pylint: disable=undefined-loop-variable
         def _cond_part():
-            rule = "<b>{}</b> ".format(class_name)
+            rule = f"<b>{class_name}</b> "
             if patt:
-                rule += "if <b>{}</b> contains <b>{}</b>".format(
-                    self.attribute.name, patt)
+                rule += f"if <b>{self.attribute.name}</b> contains <b>{patt}</b>"
             else:
                 rule += "otherwise"
             return rule
 
         def _count_part():
+            aca = "already covered above"
             if not n_matched:
-                return "all {} matching instances are already covered " \
-                       "above".format(n_total)
-            elif n_matched < n_total and patt:
-                return "{} matching instances (+ {} that are already " \
-                       "covered above".format(n_matched, n_total - n_matched)
+                if n_total == 1:
+                    return f"the single matching instance is {aca}"
+                elif n_total == 2:
+                    return f"both matching instances are {aca}"
+                else:
+                    return f"all {n_total} matching instances are {aca}"
+            elif not patt:
+                return f"{n_matched} {pl(n_matched, 'instance')}"
             else:
-                return "{} matching instances".format(n_matched)
+                m = f"{n_matched} matching {pl(n_matched, 'instance')}"
+                if n_matched < n_total:
+                    n_already = n_total - n_matched
+                    m += f" (+{n_already} that {pl(n_already, 'is|are')} {aca}"
```

Comment:
Ok, you've handled the missing `' '`, but can you handle the missing `)`

Reply:
Not yet. You must understand that developing a solution for missing spaces took years and years of programming experience. The problem of missing parentheses is at least as hard as missing spaces, so it will require an even greater moment of inspiration.

For now, I just humbly added it.